### PR TITLE
Start Kubernetes Aware Admin module with limited permissions

### PR DESCRIPTION
**Pull request by sivanov-nuodb**
**Changes**
- Make sure that KAA starts when a limited set of permissions are present in RBAC
- This addresses the backward compatibility issue described in https://github.com/nuodb/nuodb-helm-charts/issues/140

### CODE REVIEW

File: test/minikube/minikube_domain_resync_test.go [hunk 1]
```diff
@@ -465,4 +488,61 @@ func TestLoadBalancerConfigurationResync(t *testing.T) {
...
+	if os.Getenv("NUODB_LICENSE") != "ENTERPRISE" {
+		t.Skip("This test requires NuoDB 4.1.1+")
+	}
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	modifyKubeInspectorRoleInPlace(t, func(role *rbacv1.Role) {
+		// Remove "daemonsets" from the resources list
+		role.Rules[1].Resources = []string{"statefulsets", "deployments"}
```
Comment (adriansuarez) [inline test/minikube/minikube_domain_resync_test.go:503]: Also removing `pods`?

File: test/minikube/minikube_domain_resync_test.go [hunk 2]
```diff
@@ -188,6 +194,23 @@ func checkInitialMembership(t assert.TestingT, configJson string, expectedSize i
...
+	testlib.AddTeardown(testlib.TEARDOWN_ADMIN, func() { ioutil.WriteFile(inspectorRoleFile, originalData, 0644) })
+
+	output := helm.RenderTemplate(t, &helm.Options{}, testlib.ADMIN_HELM_CHART_PATH, "release-name", []string{"templates/role.yaml"})
+	roles := testlib.SplitAndRenderRole(t, output, 1)
+	modificationFunc(&roles[0])
+	roleBytes, err := yaml.Marshal(&roles[0])
+	assert.NoError(t, err)
+	err = ioutil.WriteFile(inspectorRoleFile, roleBytes, 0644)
+	assert.NoError(t, err)
+	out, _ := ioutil.ReadFile(inspectorRoleFile)
```
Comment (vegichan) [inline test/minikube/minikube_domain_resync_test.go:210]: are you overriding the file on disk? I would just create a temp file

File: test/minikube/minikube_domain_resync_test.go [hunk 3]
```diff
@@ -465,4 +488,61 @@ func TestLoadBalancerConfigurationResync(t *testing.T) {
...
 	options.SetValues["admin.lbConfig.policies.manual"] = "random(any)"
 	verifyLoadBalancer(t, namespaceName, admin0, options.SetValues)
-}
\ No newline at end of file
+}
+
+func TestKaaLimitedPermissions(t *testing.T) {
+	if os.Getenv("NUODB_LICENSE") != "ENTERPRISE" {
+		t.Skip("This test requires NuoDB 4.1.1+")
+	}
```
Comment (vegichan) [inline test/minikube/minikube_domain_resync_test.go:496]: hmm. This works, but I don't like using the license as a version guard. 

My idea was to create a new file and put in the `enterprise` build guard. Once we upgrade to say 4.1.1 as the default NuoDB instance, we can then move the whole file to say `long`.

File: test/minikube/minikube_domain_resync_test.go [hunk 4]
```diff
@@ -465,4 +488,61 @@ func TestLoadBalancerConfigurationResync(t *testing.T) {
...
+	// Verify that KAA will not register informer for PVCs
+	testlib.Await(t, func() bool {
+		return testlib.GetStringOccurrenceInLog(t, namespaceName, admin0,
+			"Informer for resource 'persistentvolumeclaims' not registered", &v12.PodLogOptions{}) == 1
+	}, 30*time.Second)
+
+	// Verify that resources that KAA have permissions for are available
+	adminStatefulSet := fmt.Sprintf("%s-nuodb-cluster0", helmChartReleaseName)
+	teDeployment := fmt.Sprintf("te-%s-nuodb-cluster0-demo", databaseHelmChartReleaseName)
+	config := testlib.GetNuoDBK8sConfigDump(t, namespaceName, admin0)
```
Comment (vegichan) [inline test/minikube/minikube_domain_resync_test.go:543]: good idea!

**Review by vegichan — APPROVED** [review]
besides minor comments, this is a great change. Thanks!

File: test/minikube/minikube_domain_resync_test.go (hunk 1 above)
Comment (sivanov-nuodb) [reply test/minikube/minikube_domain_resync_test.go:503]: The rule with index 1 is having normally 3 resources. This is removing just the `daemonsets` from the list.
 ```
- apiGroups:
  - apps
  resources:
  - deployments
  - statefulsets
  - daemonsets
  verbs:
  - get
  - list
  - watch
```

File: test/minikube/minikube_domain_resync_test.go (hunk 2 above)
Comment (sivanov-nuodb) [reply test/minikube/minikube_domain_resync_test.go:210]: As I know helm will send all files in the `templates` directory via the rendering engine. Either we should edit the file in place and revert the change or we should move the original file and revert it later. I preferred the first option as we don't need to write a lot of helper code just to move files around.

File: test/minikube/minikube_domain_resync_test.go (hunk 3 above)
Comment (sivanov-nuodb) [reply test/minikube/minikube_domain_resync_test.go:496]: That works for me. I will move the test to a new file and tag it as `enterprise`.

**Review by adriansuarez — APPROVED** [review]
Looks good.

**Pull request merged**